tests/test_fuzzy_checker.py

```diff
@@ -1,7 +1,259 @@
+    numerator: int, denominator: int, target_proportion: float
+) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+@pytest.mark.parametrize(
+    "numerator, denominator, target_proportion, match",
+    [
+        (901, 100_000, 0.05, "is significantly less than expected"),
+        (1_150, 50_000, 0.02, "is significantly greater than expected"),
+    ],
+)
+def test_fail_fuzzy_assert_proportion(
+    numerator: int, denominator: int, target_proportion: float, match: str
+) -> None:
+    with pytest.raises(AssertionError, match=match):
+        FuzzyChecker().fuzzy_assert_proportion(numerator, denominator, target_proportion)
+
+
+def test_small_sample_size_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    FuzzyChecker().fuzzy_assert_proportion(1, 10, 0.1)
+    assert "Sample size too small" in caplog.text
+
+
+def test_not_conclusive_fuzzy_assert_proportion(caplog: LogCaptureFixture) -> None:
+    # This test verifies we will pass, then be inconclusive, then fail
+    # The numbers used in this test are arbitrary but are intended to be conservative
+    # estimates of the number of iterations needed to reach each state
+    # This is to cache some of the computation for the while loop
+    fuzzy_checker = FuzzyChecker()
+    i = 1_000
+    while True:
+        caplog.clear()
+        fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+        if "is not conclusive" in caplog.text:
+            assert i > 1050
+            break
+        if i > 1_200:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+    while True:
+        caplog.clear()
+        try:
+            fuzzy_checker.fuzzy_assert_proportion(i, 10_000, 0.1)
+            assert "is not conclusive" in caplog.text
+        except AssertionError as e:
+            assert "is significantly greater" in str(e)
+            assert i > 1_100
+            break
+        if i > 1_300:
+            raise RuntimeError("Test did not reach the expected warning")
+        i += 1
+
+
+@pytest.mark.parametrize(
+    "numerator",
+    [
+        1_008,
+        10_008,
+        100_008,
+        1_000_008,
+    ],
+)
+@pytest.mark.parametrize("denominator", OBSERVED_DENOMINATORS, ids=lambda x: x)
+def test__calculate_bayes_factor(numerator: int, denominator: int) -> None:
+    if numerator > denominator:
+        pytest.skip("Numerator cannot be greater than denominator.")
+    # Parametrize rv_discrete for no bug distribution
+    # I am keeping the defaults for the bug distribution to remain 0.5 for alpha and beta
+    bug_issue_distribution = scipy.stats.betabinom(a=0.5, b=0.5, n=denominator)
+    no_bug_issue_distribution = scipy.stats.binom(p=TARGET_PROPORTION, n=denominator)
+    bayes_factor = FuzzyChecker()._calculate_bayes_factor(
+        numerator, bug_issue_distribution, no_bug_issue_distribution
+    )
+    assert isinstance(bayes_factor, float)
```

Comment:
Should we be testing these values at all? At the very least, they should be positive. But probably we want to check that they are in the right ballpark.

Reply:
@zmbc  what is a reason ballpark be? I think we should have some constraints on the outputs but they shouldn't be too constrictive.

Reply:
Bayes factors should always be positive. They should be larger than 1 when the evidence supports a bug, and less than 1 when the evidence does not support a bug. If you want to check these, you might want to handwrite some cases as opposed to doing the grid of numerators x denominators.

Reply:
@albrja it doesn't seem to me that this got implemented? Did you decide against it (or perhaps I just don't even know what to look for)

Reply:
@stevebachmeier we added additional functionality and checks here.